ui/utility.py

```diff
@@ -9,17 +9,43 @@
 import re
 import logging
 import functools
+import tarfile
+import shutil
+from datetime import datetime
 log = logging.getLogger(__name__)
 
 class FormfeedConversionException(Exception): pass
 class LinefeedConversionException(Exception): pass
 
-def find_firmware(directory):
+def update_ui(config):
+    log.info("updating UI")
+    usb_dir = config.get('files', 'usb_dir')
+    ui_file = find_ui_update(usb_dir)
+
+    log.info("looking for new ui in %s" % usb_dir)
+    install_dir = config.get('files', 'install_dir')
+    install_dir = os.path.expanduser(install_dir)
+
+    archive_dir = config.get('files', 'archive_dir')
+    archive_dir = os.path.expanduser(archive_dir)
+
+    # archive old ui
+    archive_dir += datetime.now().strftime("%Y%m%d-%H%M%S-ui")
+    log.info("archiving %s to %s" % (install_dir, archive_dir))
+    shutil.move(install_dir, archive_dir)
```

Comment:
everything is moving together (so it's not like removing all the files of a running program), it's tested and works fine. 
I'm happy to do it a different way but then we need to agree on a method to make it happen. Here is my suggestion:

* if ui update is selected, and the file exists - touch a file as a flag.
* exit.

Note that a file is touched to avoid having to deal with reading the state. As the state is pickled, it can only be opened from within the directory it was made in so we would have the same situation as in this branch.

The supervisory loop runs the main.py process and a new program called update_ui.py. If update_ui.py finds the flag it archives the old directory and unarchives the new one in its place.

Comments?